app/Domains/Contact/ManageReligion/Services/UpdateReligion.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Domains\Contact\ManageReligion\Services;
+
+use App\Interfaces\ServiceInterface;
+use App\Models\Contact;
+use App\Models\ContactFeedItem;
+use App\Models\Religion;
+use App\Services\BaseService;
+use Carbon\Carbon;
+
+class UpdateReligion extends BaseService implements ServiceInterface
+{
+    private Religion $religion;
+
+    /**
+     * Get the validation rules that apply to the service.
+     *
+     * @return array
+     */
+    public function rules(): array
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'vault_id' => 'required|integer|exists:vaults,id',
+            'author_id' => 'required|integer|exists:users,id',
+            'contact_id' => 'required|integer|exists:contacts,id',
+            'religion_id' => 'nullable|integer|exists:religions,id',
+        ];
+    }
+
+    /**
+     * Get the permissions that apply to the user calling the service.
+     *
+     * @return array
+     */
+    public function permissions(): array
+    {
+        return [
+            'author_must_belong_to_account',
+            'vault_must_belong_to_account',
+            'author_must_be_vault_editor',
+            'contact_must_belong_to_vault',
+        ];
+    }
+
+    /**
+     * Update religion of the given contact.
+     *
+     * @param  array  $data
+     * @return Contact
+     */
+    public function execute(array $data): Contact
+    {
+        $this->validateRules($data);
+
+        $this->religion = Religion::where('account_id', $data['account_id'])
+            ->findOrFail($data['religion_id']);
+
+        $this->contact->religion_id = $data['religion_id'] ? $this->religion->id : null;
+        $this->contact->save();
+
+        $this->updateLastEditedDate();
+
+        ContactFeedItem::create([
+            'author_id' => $this->author->id,
+            'contact_id' => $this->contact->id,
+            'action' => ContactFeedItem::ACTION_RELIGION_UPDATED,
+        ]);
+
+        return $this->contact;
+    }
+
+    private function updateLastEditedDate(): void
+    {
+        $this->contact->last_updated_at = Carbon::now();
+        $this->contact->save();
+    }
```

Comment:
There is laravel built-in way to do this: https://laravel.com/docs/9.x/eloquent-relationships#touching-parent-timestamps

Reply:
Ok so I didn't know and it's great.
However in our codebase, there are some time where contact is not a parent model and we still need to indicate that the contact's `updated_at` field should be updated.
This method from Laravel is a bit too _magical_ for me. I prefer to call it explicitly in the service so I know that this method explicitly needs to update the parent model.